src/app/config/urls.ts

```diff
@@ -106,11 +111,11 @@ export const LocationToPath: Record<Location, string> = {
   [Location.Dashboard]: '/tool-dashboard',
   [Location.AlignResult]: '/align/:id/:subPage?',
   [Location.Align]: '/align',
-  [Location.BlastResult]: '/blast/:id/:subPage?',
+  [Location.BlastResult]: '/blast/:namespace?/:id/:subPage?',
   [Location.Blast]: '/blast',
   [Location.PeptideSearchResult]: '/peptide-search/:id/:subPage?',
   [Location.PeptideSearch]: '/peptide-search',
-  [Location.IDMappingResult]: '/id-mapping/:id',
+  [Location.IDMappingResult]: '/id-mapping/:namespace?/:id',
```

Comment:
It shouldn't be optional for BLAST actually. What would you suggest for ID Mapping when the target db is not a namespace?

Reply:
Or I could define 2 locations:
 [Location.IDMappingResult]: '/id-mapping/:namespace/:id',
 [Location.IDMappingResult]: '/id-mapping/:id',
?

Reply:
Maybe in order to be sure that the id value will be set properly by the route match (and not end up in the namespace value in this case) you would need to specify the possible values namespace can take (see Location.Basket approach). It would allow you to have one location only with all the possible cases, and id being set correctly in all cases

Reply:
So, in the case where the target database is not a namespace, would you expect `/id-mapping/id-mapping/:id`?

Reply:
I was thinking about `/id-mapping/:namespace(uniprotkb|uniref|uniparc)?/:id` which should allow:
 - /id-mapping/uniprotkb/1234 -> namespace: uniprotkb, id: 1234
 - /id-mapping/uniref/1234 -> namespace: uniref, id: 1234
 - /id-mapping/uniparc/1234 -> namespace: uniparc, id: 1234
 - /id-mapping/1234 -> namespace: _undefined_, id: 1234